Makefile

```diff
@@ -1,9 +1,27 @@
+SHELL := bash
+.SHELLFLAGS := -eu -o pipefail -c
+.DELETE_ON_ERROR:
+
+# add easyjson source file targets here
+easyjson_sources := basm/dto.go x/xbasm/parse.go
+easyjson_generated := $(easyjson_sources:.go=_easyjson.go)
+
+# Rule to generate *_easyjson.go files
+%_easyjson.go: %.go
+	@easyjson $<
+
+# Generate all *_easyjson.go files
+generate: $(easyjson_generated)
+
+.PHONY: lint
 lint:
 	golangci-lint run --config golangci.yaml
 
-pre-pr: lint
-	$(MAKE) -C ./example clean run
+pre-pr: lint generate
```

Comment:
Isn't `pre-pr` a `.PHONY` target? By definition, it does not generate "a file with the same name as the target" (e.g., `pre-pr.out`)

Reply:
yep, you are totally right, thanks!